### PR TITLE
Dashboards rename related changes

### PR DESCRIPTION
**Pull request by hsiang9431-amzn**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Enhancement, Documentation
 
 
 2. __Github Issue # or road-map entry, if available:__
N/A

 3. __Description of changes:__
Incorporate dashboards name change to security plugin

 4. __Why these changes are required?__ 
filling gaps for dashboards name changes

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 unit tests, manual tests
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: securityconfig/internal_users.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ admin:
 kibanaserver:
   hash: "$2a$12$4AcgAt3xwOWadA5s5blL6ev39OXDNhmOesEoo33eZtrq2N0YrU3H."
   reserved: true
-  description: "Demo kibanaserver user"
+  description: "Demo opensearch dashboards server user"
```
Comment (sujithvm) [inline securityconfig/internal_users.yml:23]: We are changing the description but keeping the role name as "kibana" itself which causes confusion. Is this expected ?

Comment (sujithvm) [reply securityconfig/internal_users.yml:23]: Or should this PR be created at a later stage when roles are renamed also?

Comment (dblock) [reply securityconfig/internal_users.yml:23]: If this is not a magic word, write "OpenSearch Dashboards" with camelcase spelling, "server" is not a thing AFAIK

Comment (hsiang9431-amzn) [reply securityconfig/internal_users.yml]: fixed

File: src/main/java/org/opensearch/security/support/ModuleType.java [hunk 2]
```diff
@@ -52,7 +52,7 @@
 	REST_MANAGEMENT_API("REST Management API", "org.opensearch.security.dlic.rest.api.SecurityRestApiActions", Boolean.TRUE),
 	DLSFLS("Document- and Field-Level Security", "org.opensearch.security.configuration.SecurityFlsDlsIndexSearcherWrapper", Boolean.TRUE),
 	AUDITLOG("Audit Logging", "org.opensearch.security.auditlog.impl.AuditLogImpl", Boolean.TRUE),
-	MULTITENANCY("Kibana Multitenancy", "org.opensearch.security.configuration.PrivilegesInterceptorImpl", Boolean.TRUE),
+	MULTITENANCY("OpenSearch Dashboards  Multi-tenancy", "org.opensearch.security.configuration.PrivilegesInterceptorImpl", Boolean.TRUE),
```
Comment (palashhedau) [inline src/main/java/org/opensearch/security/support/ModuleType.java:55]: nit: extra space between Dashboards and Multi-tenancy

File: src/main/java/org/opensearch/security/support/ModuleType.java [hunk 3]
```diff
@@ -69,7 +69,7 @@
 	CUSTOM_HTTP_AUTHENTICATOR("Custom HTTP authenticator", null, Boolean.TRUE),
 	CUSTOM_AUTHENTICATION_BACKEND("Custom authentication backend", null, Boolean.TRUE),
 	CUSTOM_AUTHORIZATION_BACKEND("Custom authorization backend", null, Boolean.TRUE),
-	CUSTOM_INTERCLUSTER_REQUEST_EVALUATOR("Intercluster Request Evaluator", null, Boolean.FALSE),
+	CUSTOM_INTERCLUSTER_REQUEST_EVALUATOR("Inter-cluster Request Evaluator", null, Boolean.FALSE),
```
Comment (palashhedau) [inline src/main/java/org/opensearch/security/support/ModuleType.java:72]: why this?

Comment (hsiang9431-amzn) [reply src/main/java/org/opensearch/security/support/ModuleType.java:72]: my spelling check complains about it

File: src/main/java/org/opensearch/security/support/ModuleType.java (hunk 2 above)
Comment (hsiang9431-amzn) [reply src/main/java/org/opensearch/security/support/ModuleType.java]: fixed

**Review by palashhedau — APPROVED** [review]

**Pull request closed without merging**